flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSourceUtils.java

```diff
@@ -90,13 +90,18 @@ public static RelNode convertDataStreamToRel(
         final DynamicTableSource tableSource =
                 new ExternalDynamicSource<>(
                         identifier, dataStream, physicalDataType, isTopLevelRecord, changelogMode);
+        final FlinkStatistic statistic =
+                FlinkStatistic.builder()
+                        // this is a temporary solution, FLINK-15123 will resolve this
```

Comment:
(The good ol' two-year-and-counting temporary solution… :-))

Reply:
yes, and then you wonder why your primary keys are lost if you don't have statistics defined